images/replication-merger/start.py

```diff
@@ -0,0 +1,313 @@
+        bottom_dir = parse_integer_to_directory_number(int(i / 1000))
+        state_value = parse_integer_to_directory_number(int(i % 1000)) 
+        first_state_file = os.path.join(time_unit.sub_time_unit_data_dir, top_dir, bottom_dir, f"{state_value}.state.txt")
+        first_state_file_content = get_file_from_server(first_state_file)
+        if first_state_file_content == -1:
+            continue
+        log.info(f"found first state file in {first_state_file}")
+        set_state_file_content(time_unit.get_path(), first_state_file_content)
+        return True
+    return False
+
+def set_state_file_content(time_unit_dir, state_content):
+    '''
+    Overwrites a timeunit state file in two levels, it's own replication and it's base upon it merges by.
+    Used when initializing a time unit environment
+    Args:
+        time_unit_dir (str): the replication directory of the time unit
+        state_content (str): the state file content that will be used in the override
+    '''
+    time_unit_state = os.path.join(time_unit_dir, STATE_FILE)
+    overwrite_file(time_unit_state, state_content)
+    time_unit_data_state = os.path.join(time_unit_dir, DATA_FOLDER, STATE_FILE)
+    overwrite_file(time_unit_data_state, state_content)
+
+def init_time_unit_merge(time_unit):
+    """
+    Initializes the time unit replication merge environment if not already exists
+    Args:
+        time_unit (TimeUnit): the time unit to init its merge env
+    """
+    time_unit_state_path = os.path.join(time_unit.get_path(), STATE_FILE)
+    if not os.path.isfile(time_unit_state_path):
+        log.info('initializing merge replication environment')
+        os.makedirs(os.path.join(time_unit.get_path(), DATA_FOLDER), exist_ok=True)
+        copy_time_unit_config(time_unit.name)
+        found = http_fetch_starting_state(time_unit)
+        if not found:
+            log_and_exit('could not find starting state from the based on time unit')
+
+def copy_time_unit_config(time_unit_name):
+    """
+    Copies the time unit configuration instead of default one generated by osmosis
+    Args:
+        time_unit_name (str): the name of the time unit
+    """
+    time_unit_dir_path = os.path.join(REPLICATION_DIR, time_unit_name)
+    time_unit_config_src = os.path.join(CONFIG_DIR, f"{time_unit_name}-config.txt")
+    time_unit_config_dst = os.path.join(time_unit_dir_path, CONFIGURATION_FILE)
+    shutil.copy(time_unit_config_src, time_unit_config_dst)
+
+def load_merge_configuration():
+    """
+    Sets the wanted TimeUnit to be merged on the job
+    TimeUnit will be initialized with its configuration file
+    """
+    if TIME_UNIT_TO_MERGE not in Time_Unit._member_names_:
+        log.error(f"time unit {TIME_UNIT_TO_MERGE} cannot be merged.")
+        raise ValueError()
+
+    global time_unit
+    time_unit = TimeUnit(TIME_UNIT_TO_MERGE, os.path.join(CONFIG_DIR, f"{TIME_UNIT_TO_MERGE}-config.txt"))
```

Comment:
why do you use global ?

Reply:
> Python assumes that any name that is assigned to, anywhere within a function, is local to that function and will create a local variable that will shadow the global variable of the same name unless explicitly told otherwise. 
If it is only reading from a name, and the name doesn't exist locally, it will try to look up the name in any containing scopes (e.g. the module's global scope).

Reply:
What I meant is why did you decide to use `global`?

it is considered a bad practice.

Reply:
removed the global time_unit variable.